table_schema_cache.go

```diff
@@ -268,6 +266,14 @@ func (t *TableSchema) paginationKeyColumn(cascadingPaginationColumnConfig *Casca
 	return paginationKeyColumn, err
 }
 
+func (t *TableSchema) fallbackPaginationColumnName(cascadingPaginationColumnConfig *CascadingPaginationColumnConfig) string {
```

Comment:
Here you are defining what is called a private method on `*TableSchema` except that inside this method you are not leveraging anything `*TableSchema`-related.

As such, it would be better to define it as a method on `*CascadingPaginationColumnConfig` instead. Therefore, it would be better to define this as a public method inside `config` instead such as: https://github.com/Shopify/ghostferry/blob/9394cbf6114555ed635ed35f9dc0874d40773fd8/config.go#L284

Next, the more idiomatic return type for such a method would be a tuple: `(string, bool)`